modules/markup/markdown/goldmark.go

```diff
@@ -151,6 +151,16 @@ func (g *ASTTransformer) Transform(node *ast.Document, reader text.Reader, pc pa
 					v.AppendChild(v, newChild)
 				}
 			}
+		case *ast.Text:
+			if v.SoftLineBreak() && !v.HardLineBreak() {
+				renderMetas := pc.Get(renderMetasKey).(map[string]string)
+				mode := renderMetas["mode"]
+				if mode != "document" {
+					v.SetHardLineBreak(setting.Markdown.EnableHardLineBreakInComments)
```

Comment:
Should this be set on each node? 🤔

Reply:
yes.

The text nodes are rendered by: 

https://github.com/yuin/goldmark/blob/a302193b064875a8af8cd241985cb26574f37408/renderer/html/html.go#L602-L623

I cannot adjust the renderer as you do not have the parsercontext at that point.